stackit/internal/services/iaas/server/resource.go

```diff
@@ -422,13 +439,21 @@ func (r *serverResource) Create(ctx context.Context, req resource.CreateRequest,
 	}
 
 	serverId := *server.Id
-	server, err = wait.CreateServerWaitHandler(ctx, r.client, projectId, serverId).WaitWithContext(ctx)
+	_, err = wait.CreateServerWaitHandler(ctx, r.client, projectId, serverId).WaitWithContext(ctx)
 	if err != nil {
 		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating server", fmt.Sprintf("server creation waiting: %v", err))
 		return
 	}
 	ctx = tflog.SetField(ctx, "server_id", serverId)
 
+	// Get Server with details
+	serverReq := r.client.GetServer(ctx, projectId, serverId)
+	serverReq = serverReq.Details(true)
+	server, err = serverReq.Execute()
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error get created server with details", fmt.Sprintf("get server: %v", err))
```

Comment:
Generally we were keeping the title of the Terraform error uniform for each function

E.g. in this case, since this is part of the server creation, it would look something like:

```suggestion
		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating server", fmt.Sprintf("get server details: %v", err))
```